app/javascript/hooks/useTimeout.ts

```diff
@@ -0,0 +1,22 @@
+import { useRef, useCallback } from 'react';
+
+export const useTimeout = () => {
+  const timeoutRef = useRef<ReturnType<typeof setTimeout>>();
+
+  const set = useCallback((callback: () => void, delay: number) => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
+    timeoutRef.current = setTimeout(callback, delay);
+  }, []);
+
+  const cancel = useCallback(() => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+      timeoutRef.current = undefined;
+    }
+  }, []);
```

Comment:
Should we have a `useEffect` here to cancel the timeout when the calling component is unmounted? I am not sure there are cases when you want the callback to be called if the component creating it is no longer there